app/models/reviewable_score.rb

```diff
@@ -0,0 +1,86 @@
+class ReviewableScore < ActiveRecord::Base
+  belongs_to :reviewable
+  belongs_to :user
+  belongs_to :reviewed_by, class_name: 'User'
+  belongs_to :meta_topic, class_name: 'Topic'
+
+  def self.statuses
+    @statuses ||= Enum.new(
+      pending: 0,
+      agreed: 1,
+      disagreed: 2,
+      ignored: 3
+    )
+  end
+
+  def self.score_transitions
+    {
+      approved: statuses[:agreed],
+      rejected: statuses[:disagreed],
+      ignored: statuses[:ignored]
+    }
+  end
+
+  # Generate `pending?`, `rejected?`, etc helper methods
+  statuses.each do |name, id|
+    define_method("#{name}?") { status == id }
+    self.class.define_method(name) { where(status: id) }
+  end
+
+  def score_type
+    Reviewable::Collection::Item.new(reviewable_score_type)
+  end
+
+  def took_action?
+    take_action_bonus > 0
+  end
+
+  # A user's flag score is:
+  #   1.0 + trust_level + user_accuracy_bonus
+  #   (trust_level is 5 for staff)
+  def self.user_flag_score(user)
+    1.0 + (user.staff? ? 5.0 : user.trust_level.to_f) + user_accuracy_bonus(user)
+  end
+
+  # A user's accuracy bonus is:
+  #   if 5 or less flags => 0.0
+  #   if > 5 flags => (agreed flags / total flags) * 5.0
+  def self.user_accuracy_bonus(user)
+    user_stat = user&.user_stat
+    return 0.0 if user_stat.blank?
+
+    total = (user_stat.flags_agreed + user_stat.flags_disagreed + user_stat.flags_ignored).to_f
+    return 0.0 if total <= 5
+
+    (user_stat.flags_agreed / total) * 5.0
+  end
+
+  def reviewable_conversation
+    return if meta_topic.blank?
+    Reviewable::Conversation.new(meta_topic)
+  end
+
+end
+
+# == Schema Information
+#
+# Table name: reviewable_scores
+#
+#  id                    :bigint(8)        not null, primary key
+#  reviewable_id         :integer          not null
+#  user_id               :integer          not null
+#  reviewable_score_type :integer          not null
+#  status                :integer          not null
+#  score                 :float            default(0.0), not null
+#  take_action_bonus     :float            default(0.0), not null
+#  reviewed_by_id        :integer
+#  reviewed_at           :datetime
+#  meta_topic_id         :integer
+#  created_at            :datetime         not null
+#  updated_at            :datetime         not null
+#
+# Indexes
```

Comment:
I feel like most of the `*_id` here will benefit from an index.

Reply:
There is a performance penalty on indexing columns where inserts become slower. I would rather show that we are using those columns for a particular thing rather than adding indexes on everything. I am not as certain that we will filter a score based on who reviewed it, for example. You'll almost always want to show a score associated with a reviewable (which has an index.)

Reply:
I guess I’ve always preferred to add indexes to foreign key columns since they are very commonly used in filters and we tend to forget about the index when adding filters in the future. My view is that the overhead on insertion is there but not expensive though. Most of the problems I’ve seen with slow DB queries are usually due to missing indexes on read queries and almost never on write. However, whether we have the additional indexes here or not is not a show stopper :P

Reply:
This sounds like a dev topic to me "Should we index all foreign key columns?" :)

For now I've been watching the perf closely (and will continue to do so after merge) and I believe we will catch any missing indexes. If you want to open that dev topic was can make it a bigger discussion which GitHub is not ideal for!